src/methopt/conjugate_direction_method.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+
+def conjugate_direction_method(
+    Q,
+    b,
+    x0,
+    max_iterations_count=1000,  # todo: I think it isn't needed
+    iteration_callback=None,
+):
+    # f(x) = 0.5 (Qx, x) + (b, x)
+    if iteration_callback is None:
+        iteration_callback = lambda **kwargs: ()
+
+    if np.all(b == 0):
+        iteration_callback(x=b, iteration_no=0)
+        return b
+
+    iteration_callback(x=x0, iteration_no=0)
+    w1 = -Q @ x0 - b
+    p1 = w1
+    if np.linalg.norm(p1) == 0:
+        return x0
+    h1 = np.dot(p1, p1) / np.dot(Q @ p1, p1)
+    x1 = x0 + h1 * p1
+
+    x_prev = x1
+    w_prev = w1
+    p_prev = p1
+    h_prev = h1
+    for k in range(1, max_iterations_count):
+        wk = w_prev - h_prev * (Q @ p_prev)
+        yk = np.dot(Q @ p_prev, wk) / np.dot(Q @ p_prev, p_prev)
+        pk = wk - yk * p_prev
+        if np.linalg.norm(pk) == 0:
+            return x_prev
+        hk = np.dot(wk, pk) / np.dot(Q @ pk, pk)
+        xk = x_prev + hk * pk
+        iteration_callback(x=xk, iteration_no=k)
+
+        p_prev = pk
+        w_prev = wk
+        x_prev = xk
+        h_prev = hk
+    return x_prev
```

Comment:
I want to have a number of iterations returned, as well as number of arithmetic ops performed (for the method comparison implementation). The former can be done with iteration callbacks in theory, but not the latter. You can maintain two extra counters for that and return `x_prev, iterations_count, op_count`

Reply:
It doesn't seem clear from the task description what an arithmetic operation is, or rather what we want to consider an arithmetic operation. Once we define it, we can make another type of callback which would be less of a wanky design than multiple return values. 
On the other hand it should not matter as much in the project of this scale, we it's a matter of Nastya's preference. 

However, it's still important to clarify what kind of arithmetic operations we should count

Reply:
It kind of feels to me that it would be really hard to count by hand all arithmetic operations done by numpy and in other complex formulae, so maybe you can make a number wrapper class that increments a counter for all arithmetic operations on it :thinking: